app/components/pay-page/choose-membership-plan-modal.ts

```diff
@@ -0,0 +1,108 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import fade from 'ember-animated/transitions/fade';
+import { inject as service } from '@ember/service';
+import Store from '@ember-data/store';
+import window from 'ember-window-mock';
+import * as Sentry from '@sentry/ember';
+import type RegionalDiscountModel from 'codecrafters-frontend/models/regional-discount';
+import type PromotionalDiscountModel from 'codecrafters-frontend/models/promotional-discount';
+import type AuthenticatorService from 'codecrafters-frontend/services/authenticator';
+
+export interface PricingPlan {
+  id: 'quarterly' | 'yearly' | 'lifetime';
+  title: string;
+  priceInDollars: number;
+  validityInMonths?: number; // undefined for lifetime
+}
+
+export const PRICING_PLANS: PricingPlan[] = [
+  { id: 'quarterly', title: '3 month pass', priceInDollars: 120, validityInMonths: 3 },
+  { id: 'yearly', title: '1 year pass', priceInDollars: 360, validityInMonths: 12 },
+  { id: 'lifetime', title: 'Lifetime pass', priceInDollars: 1490 },
+];
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    activeDiscountForYearlyPlan?: PromotionalDiscountModel | null;
```

Comment:
Let's use either `?` or `| null`, not both - the ultimate signature here would be `Model | null | undefined` if we did this, better to restrict to one so it's easier to deal with within the component.